core/store/src/trie/mod.rs

```diff
@@ -637,18 +658,45 @@ impl Trie {
         };
     }
 
-    // Prints the trie leaves starting from the state root node, up to max_depth depth.
-    // This method can only iterate starting from the root node and it only prints the
-    // leaf nodes but it shows output in more human friendly way.
-    pub fn print_recursive_leaves(&self, f: &mut dyn std::io::Write, max_depth: u32) {
-        let iter = match self.iter_with_max_depth(max_depth as usize) {
+    /// Prints the trie leaves starting from the state root node, up to max_depth depth.
+    /// This method can only iterate starting from the root node and it only prints the
+    /// leaf nodes but it shows output in more human friendly way.
+    /// Optional arguments `limit` and `record_type` limits the output to at most `limit`
+    /// entries and shows trie nodes of `record_type` type only.
+    /// `from` and `to` can be used skip leaves with `AccountId` less than `from` or greater than `to`.
+    pub fn print_recursive_leaves(
+        &self,
+        f: &mut dyn std::io::Write,
+        max_depth: u32,
+        limit: Option<u32>,
+        record_type: Option<u8>,
```

Comment:
nit: use an enum, ideally the one from state record

Reply:
It would make `core/store` dependent on `tools/state-viewer`, wdyt?

Reply:
If I move it to `core/store` then Rust will not allow me to define `impl clap::ValueEnum` for it within `tools/state-viewer`. Rust requires that those are in the same crate.